cluster-autoscaler/core/static_autoscaler_test.go

```diff
@@ -992,115 +1108,123 @@ func TestStaticAutoscalerRunOnceWithFilteringOnUpcomingNodesEnabledNoScaleUp(t *
-			ret := onScaleDownMock.ScaleDown(id, name)
-			deleteFinished <- true
-			return ret
-		})
-	provider.AddNodeGroup("ng1", 1, 10, 1)
-	provider.AddNode("ng1", n1)
-
-	ng1 := reflect.ValueOf(provider.GetNodeGroup("ng1")).Interface().(*testprovider.TestNodeGroup)
-	assert.NotNil(t, ng1)
-	assert.NotNil(t, provider)
-
-	// Create context with mocked lister registry.
+	bypassedScheduler := "bypassed-scheduler"
+	nonBypassedScheduler := "non-bypassed-scheduler"
 	options := config.AutoscalingOptions{
 		NodeGroupDefaults: config.NodeGroupAutoscalingOptions{
 			ScaleDownUnneededTime:         time.Minute,
 			ScaleDownUnreadyTime:          time.Minute,
 			ScaleDownUtilizationThreshold: 0.5,
 			MaxNodeProvisionTime:          10 * time.Second,
 		},
-		EstimatorName:             estimator.BinpackingEstimatorName,
-		ScaleDownEnabled:          true,
-		MaxNodesTotal:             10,
-		MaxCoresTotal:             10,
-		MaxMemoryTotal:            100000,
-		IgnoreSchedulerProcessing: true,
-		IgnoredSchedulers: map[string]bool{
+		EstimatorName:    estimator.BinpackingEstimatorName,
+		ScaleDownEnabled: true,
+		MaxNodesTotal:    10,
+		MaxCoresTotal:    10,
+		MaxMemoryTotal:   100000,
+		BypassedSchedulers: map[string]bool{
 			apiv1.DefaultSchedulerName: true,
-			ignoredScheduler:           true,
+			bypassedScheduler:          true,
 		},
 	}
-	processorCallbacks := newStaticAutoscalerProcessorCallbacks()
 
-	context, err := NewScaleTestAutoscalingContext(options, &fake.Clientset{}, nil, provider, processorCallbacks, nil)
-	assert.NoError(t, err)
-
-	setUpScaleDownActuator(&context, options)
+	n1 := BuildTestNode("n1", 1000, 1000)
+	SetNodeReadyState(n1, true, time.Now())
 
-	listerRegistry := kube_util.NewListerRegistry(allNodeLister, readyNodeLister, allPodListerMock,
-		podDisruptionBudgetListerMock, daemonSetListerMock,
-		nil, nil, nil, nil)
-	context.ListerRegistry = listerRegistry
+	p1 := BuildTestPod("p1", 600, 100)
+	p1.Spec.NodeName = "n1"
+	p2 := BuildTestPod("p2", 100, 100, AddSchedulerName(bypassedScheduler))
+	p3 := BuildTestPod("p3", 600, 100, AddSchedulerName(apiv1.DefaultSchedulerName)) // Not yet processed by scheduler, default scheduler is ignored
+	p4 := BuildTestPod("p4", 600, 100, AddSchedulerName(bypassedScheduler))          // non-default scheduler & ignored, expects a scale-up
+	p5 := BuildTestPod("p5", 600, 100, AddSchedulerName(nonBypassedScheduler))
 
-	clusterStateConfig := clusterstate.ClusterStateRegistryConfig{
-		OkTotalUnreadyCount: 1,
+	testCases := map[string]testCase{
+		"Unprocessed pod with bypassed scheduler doesn't cause a scale-up when there's capacity": {
+			pods:               []*apiv1.Pod{p1, p2},
+			podListerCallTimes: 2,
+			nodeGroups: []*nodeGroup{{
+				name:  "ng1",
+				min:   1,
+				max:   10,
+				nodes: []*apiv1.Node{n1},
+			}},
+			daemonSetListerCallTimes: 1,
+			pdbListerCallTimes:       1,
+			expectedScaleUps:         []*scaleCall{},
+			now:                      time.Now(),
```

Comment:
Each `time.Now()` call returns a slightly different timestamp, which leads me to think we don't really need to include all these values in each test case?

Reply:
Yes it was made so that I'm consistent with other tests so the original "testAutoscaler" wouldn't need much updates to be used for other tests, updated now